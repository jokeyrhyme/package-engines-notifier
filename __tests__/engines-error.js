/* @flow */
'use strict'

const chalk = require('chalk')

const enginesError = require('../index.js').enginesError

const oldProcessVersion = process.version
afterEach(() => {
  process.version = oldProcessVersion
})

const pkg = {
  name: 'my-package',
  engines: {
    node: '>=4'
  }
}

test('{ node: ">=4" vs "v0.12.0" }', () => {
  process.version = 'v0.12.0'
  const err = enginesError({ pkg })
  expect(err).toBeInstanceOf(Error)
  expect(err && err.data.engine).toBe('node')
  expect(err && err.data.current).toBe('v0.12.0')
  expect(err && err.data.required).toBe('>=4')
  expect(err && chalk.stripColor(err.toString())).toMatchSnapshot()
})

test('{ node: ">=4" vs "v6.9.0" }', () => {
  process.version = 'v6.9.0'
  const err = enginesError({ pkg })
  expect(err).toBeNull()
})
