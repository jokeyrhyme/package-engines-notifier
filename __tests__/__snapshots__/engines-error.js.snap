exports[`test { node: ">=4" vs "v0.12.0" } 1`] = `
"my-package requires node >=4 
current node version is v0.12.0"
`;
